proposals/0000-slurp.rst

```diff
@@ -0,0 +1,454 @@
+   to diverge, which would make life harder for users.
+-  Source-code mining tools (e.g search, discovery, etc) would have two
+   places and perhaps two metadata formats to handle.
+-  Since each would doubtless seek to mirror packages in the other,
+   there would be much duplication and (more damagingly)
+   near-duplication, which would make life harder for both users and
+   tooling.
+
+Perhaps, after a long time, one ecosystem or another will become
+de-facto dominant. But there would be considerable pain to thousands of
+Haskell users meanwhile.
+
+That is terrible! We have so much in common.
+
+-  Everyone wants Haskell to succeed and be widely used
+-  Everyone wants Haskell users to have a frictionless experience of
+   using libraries written by others
+-  Everyone is trying hard to do the Right Thing
+
+That makes it tantalising that we are having difficulty finding a
+solution. But it also makes us optimistic that we can find one if we
+work together.
+
+Appendix A: SLURP governance and terms of use
+---------------------------------------------
+
+The below is an example policy document. It should be discussed
+independently of the technical proposal above: none of the below imposes
+any additional constraints on the implementation.
+
+Governance
+~~~~~~~~~~
+
+SLURP will be governed by the *SLURP trustees*.
+
+The SLURP trustees include representatives from the various stakeholders
+whose composition will change as the ecosystems around SLURP changes.
+Initially, we propose the SLURP trustees to be comprised of (1) one
+representative of the Hackage maintainers, (2) one representative of the
+Stackage curators, and (3) one member-at-large from the general Haskell
+community. The number of trustees can change over time. The
+member-at-large will chair the group of trustees.
+
+
+Changes to the composition of the SLURP trustees are decided by the SLURP trustees and need to be unanimous. All other decisions are made by simple majority vote if there is a dispute among the trustees.
+This governance mechanism is, by design, lightweight; for example, the number of trustees is small.  If, in the light of experience, it seems that a more substantial mechanism is needed, we can revisit the issue as a community.
+
+Terms of use
+~~~~~~~~~~~~
+
+Users of SLURP agree to abide by the rules stated in the following. IP
+addresses or IP ranges of users who repeatedly violate the terms of use
+may be blocked. This decision is made by the SLURP trustees.
+
+§1 Valid SLURP entries
+^^^^^^^^^^^^^^^^^^^^^^
+
+Any entry added with ``PUT /package/:pkgname`` that maps the new package
+name *pkgname* to the URL *url* must abide by the following
+restrictions:
+
+-  Both the *pkgname* and the *url* must not already exist in the
+   registry.
+-  The *pkgname* must not be unlawful, threatening, abusive, libelous,
+   defamatory, obscene, offensive, indecent, pornographic, profane, or
+   otherwise objectionable.
```

Comment:
Does this need be codified? Will it prohibit
{{{
acme-cutegirl: library and program: Maybe gives you a cute girl
acme-missiles: library: Cause serious international side effects.
}}}
which is currently on hackage?

I recommend to simply avoid codifying that, state that “trusteeds have the power to remove entries that abuse the system” and leave it to human common sense.